src/main/java/com/google/devtools/build/lib/query2/query/PathLabelVisitor.java

```diff
@@ -273,13 +281,30 @@ private void visit(Target from, Attribute attribute, Target target)
         visitAspectsIfRequired(from, attribute, target);
       }
 
-      if (!visited.add(target)) {
-        // We've been here before.
-        return;
+      if (visited.add(target)) {
+        visitEdgesOfTarget(target);
       }
+    }
 
-      LabelVisitationUtils.<InterruptedException, NoSuchThingException>visitTargetExceptionally(
-          target, edgeFilter, this::enqueue);
+    private void visitEdgesOfTarget(Target target)
+        throws InterruptedException, NoSuchThingException {
+      try {
+        LabelVisitationUtils.visitTarget(
+            target,
+            edgeFilter,
+            (from, attribute, label) -> {
+              try {
+                enqueue(from, attribute, label);
+              } catch (InterruptedException | NoSuchThingException e) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: InterruptedException is ignored. This can delay thread shutdown and clear the thread’s interrupt status. Only code that implements a thread’s interruption policy can swallow an interruption request.

**Fix**: Rethrow the InterruptedException or reinterrupt the current thread using *Thread.currentThread().interrupt()* so that higher-level interrupt handlers can function correctly.
 If you are wrapping the InterruptedException inside a RuntimeException, call *Thread.currentThread().interrupt()* before throwing the RuntimeException.

Learn more about [interrupts](https://docs.oracle.com/javase/tutorial/essential/concurrency/interrupt.html) and [dealing with InterruptedException](https://www.ibm.com/developerworks/java/library/j-jtp05236/index.html?ca=drs-#N10187)